### PR TITLE
Issue #16 Changed Builder to use environment for tracer creation

### PR DESCRIPTION
**Pull request by mwenk-cc**

### CODE REVIEW

File: opentracing-spring-jaeger-starter/src/main/java/io/opentracing/contrib/java/spring/jaeger/starter/JaegerAutoConfiguration.java [hunk 1]
```diff
@@ -65,8 +65,10 @@
   public io.opentracing.Tracer tracer(Sampler sampler,
                                       Reporter reporter) {
 
-    final JaegerTracer.Builder builder =
-        new JaegerTracer.Builder(serviceName)
+    // Serive name needs to be a system property to be picked up because of how fromEnv() works.
+    System.setProperty("JAEGER_SERVICE_NAME", serviceName);
```
Comment (geoand) [inline opentracing-spring-jaeger-starter/src/main/java/io/opentracing/contrib/java/spring/jaeger/starter/JaegerAutoConfiguration.java]: @pavolloffay Is this necessary?

File: opentracing-spring-jaeger-starter/src/main/java/io/opentracing/contrib/java/spring/jaeger/starter/JaegerAutoConfiguration.java [hunk 2]
```diff
@@ -65,8 +65,10 @@
   public io.opentracing.Tracer tracer(Sampler sampler,
                                       Reporter reporter) {
 
-    final JaegerTracer.Builder builder =
-        new JaegerTracer.Builder(serviceName)
+    // Serive name needs to be a system property to be picked up because of how fromEnv() works.
+    System.setProperty("JAEGER_SERVICE_NAME", serviceName); 
+
+    final JaegerTracer.Builder builder = io.jaegertracing.Configuration.fromEnv().getTracerBuilder()
```
Comment (geoand) [inline opentracing-spring-jaeger-starter/src/main/java/io/opentracing/contrib/java/spring/jaeger/starter/JaegerAutoConfiguration.java]: I really don't want to do this in an uncontrolled manner. If this is absolutely needed, it should be controlled by a boolean property that is off by default

**Comment by geoand** [comment]
Thank you @mwenk-cc for your PR!

Please find my comments in the review

File: opentracing-spring-jaeger-starter/src/main/java/io/opentracing/contrib/java/spring/jaeger/starter/JaegerAutoConfiguration.java (hunk 1 above)
Comment (geoand) [reply opentracing-spring-jaeger-starter/src/main/java/io/opentracing/contrib/java/spring/jaeger/starter/JaegerAutoConfiguration.java]: @objectiser Thanks! I will schedule an update to `0.31`.

**Comment by mwenk-cc** [comment]
After discussion this internally, we can't resolve creating the Tracer from environment without creating the sampler/reporter the same way.  However, doing that seems to make this very messy, especially with the fact the code would need to allow for the bean override, but for the default bean would use the env created beans.   So closing this out.

**Comment by geoand** [comment]
@mwenk-cc Thats is exactly why we haven't used the environment variables and `Configuration.fromEnv()`.
What we have done is to try and provide as much functionality as possible in a way that is idiomatic to Spring

**Pull request closed without merging**